client/app/pages/queries/hooks/useDataSourceSchema.js

```diff
@@ -1,36 +1,39 @@
-import { reduce } from "lodash";
+import { reduce, has } from "lodash";
 import { useCallback, useEffect, useRef, useState, useMemo } from "react";
 import { axios } from "@/services/axios";
 import DataSource, { SCHEMA_NOT_SUPPORTED } from "@/services/data-source";
 import notification from "@/services/notification";
 
 function sleep(ms) {
-  return new Promise((resolve) => setTimeout(resolve, ms));
+  return new Promise(resolve => setTimeout(resolve, ms));
 }
 
 function getSchema(dataSource, refresh = undefined) {
   if (!dataSource) {
     return Promise.resolve([]);
   }
 
-  const fetchSchemaFromJob = (data) => {
-    return sleep(1000).then(() => {
```

Comment:
I also removed the first request sleeping time, from a few tests a good portion of the results would be done already without the sleeping time.